### ISSUE TITLE
What do Run 3 Lensed Quasars look like?

### ISSUE THREAD

**Issue by jbkalmbach** [LSSTDESC/Twinkles#424]
Now that we have images we should take a look at some of the new systems.

**Comment by drphilmarshall** [comment]
@jbkalmbach and I worked on this with @SimonKrughoff at the SLAC2017 Hack Day - check out our [notebook](https://gist.github.com/jbkalmbach/70d4e266f778b56e76e2e2cfa1ca5f3b)! 

Here's our main result: 4 DIASources (cyan) that we spatially associated with the true position of a double-image lens system, whose lens G and images A and B are marked with an orange circle and blue/cyan crosses respectively:
 
![image](https://cloud.githubusercontent.com/assets/710903/23224549/b1daa54a-f8e3-11e6-9427-8948671e287d.png)

As you can see, the DM detections seem a bit skew-wiff. Bryce, we need to look at some more systems, which means generalizing this notebook to use a `def` that takes a `twinkles_id` as input. The separation into images A and B won't be possible, so let's just plot DIASources as cyan circles and true image positions as blue crosses (and true lens positions as orange crosses).

**Comment by SimonKrughoff** [comment]
It would be very interesting to see these plotted on top of the image.

**Comment by drphilmarshall** [comment]
Good idea. You were close to being able to do that, I think, Bryce. At what
point should we switch back to upgrading the Monitor, though?

**Comment by jbkalmbach** [comment]
![run3_image_bare](https://cloud.githubusercontent.com/assets/4755090/23448787/ddce3ee0-fe07-11e6-834e-93fb56c93f2b.png)

<img width="654" alt="run3_image" src="https://cloud.githubusercontent.com/assets/4755090/23449018/371c47a2-fe09-11e6-8384-64cd4b0122da.png">

**Comment by jbkalmbach** [comment]
![run3_image_2_bare](https://cloud.githubusercontent.com/assets/4755090/23449652/5e6adef6-fe0c-11e6-99f6-9b2a07d434f4.png)
![run3_image_2_obs](https://cloud.githubusercontent.com/assets/4755090/23449653/5e849986-fe0c-11e6-9633-3f4b6b7e85d7.png)
![run3_image_2_true](https://cloud.githubusercontent.com/assets/4755090/23449654/5e94275c-fe0c-11e6-8d98-52292a27a3e7.png)

**Comment by jbkalmbach** [comment]
It looks like there are objects where they should be and that there is something wrong in the detections we are getting back in the catalog.

**Comment by jbkalmbach** [comment]
Ok, the coordinate issue is sorted out and Scott has just issued a [PR](https://github.com/lsst/sims_catUtils/pull/77) in sims_catUtils for a method that will make it easier to convert between phosim and ICRS coordinates in the future.

![run3_image_2_bare](https://cloud.githubusercontent.com/assets/4755090/23491259/f2a6bc16-feb1-11e6-99c6-1147d311d3d1.png)


![run3_image_2_truth](https://cloud.githubusercontent.com/assets/4755090/23491249/e6799a4e-feb1-11e6-882b-68bf8b24102b.png)

**Comment by jbkalmbach** [comment]
And here's a difference image. The four rings are all DIA Sources within 2 arcseconds of the lens galaxy location. It looks like the AGN pop out in the difference image but aren't getting found.

![run3_image_2_diff](https://cloud.githubusercontent.com/assets/4755090/23491638/4ff77dae-feb4-11e6-9ae5-abaf9ca444cb.png)

**Comment by drphilmarshall** [comment]
Interesting! And also bad news as it stands :-) Thanks Bryce.

So in that particular epoch you're showing, I think we can all agree that
the stack *should* have found 4 DIASources - but it hasn't. Is there some
setting that would prevent the DIASource characterizer from "seeing" nearby
features, @SimonKrughoff? Is there a segmentation map we can inspect to see
what happened during DIASource detection and characterization?

**Comment by rbiswas4** [comment]
I thought this was good news, in that the difference image is working correctly (except for the red galaxy) and so while there is obviously a bug in recording the results, it would seem that would be relatively easy to fix to book keeping. But I have made many guesses here about what exactly is shown, and am now worried that I am misunderstanding the scenario.

@jbkalmbach Is there a record of the signifcance SNR of the difference image?  To my eyes this looks like a dream result which would be superhigh SNR in 10s ?

One thing I am puzzled by is the image D (obs), which I was thinking of as the max in Signal for the diffim sitting on C. Is that something you plotted up by hand from the diffim by looking or is the association coming from some other source ?

**Comment by rbiswas4** [comment]
One more question: I was assuming that the recorded result for these objects on this epoch was  no objects. Is this true?

Also a request: Can we see the 'template' that was used for differencing in this region?

**Comment by SimonKrughoff** [comment]
@jbkalmbach it's fantastic to see the lens showing up so well in the diffim.  The fact that they are not being detected is either 1) they are being characterized as a dipole because the galaxy is not subtracting all the way or 2) they are all blended.  I'm a little puzzled why the galaxy residual shows up, but it's possible that's in the noise.  I'm also puzzled why at least the image toward the top does not separate out.

Can you give me the visit number for that realization and I can produce a segment map.

**Comment by rbiswas4** [comment]
@SimonKrughoff For dipole characterization, is the some part being negative not enforced? And if they are blended, should they show up as at least one source instead of 4?

**Comment by SimonKrughoff** [comment]
@rbiswas4 I don't believe that the dipole characterization requires that there be exactly one negative and one positive going lobe.  I think it's at least one negative and one positive peak.  I could be wrong though.

@jbkalmbach Thanks.

**Comment by rbiswas4** [comment]
@SimonKrughoff I see ... so near the lens is what you think is negative enough to trigger dipole characterization. Ok thanks.

**Comment by jbkalmbach** [comment]
Just for completeness I went and looked at the 'src' catalog and it seems it only detects the lens galaxy (apparently there are two 'src's but they are in exactly the same spot).

![run3_image_2_src](https://cloud.githubusercontent.com/assets/4755090/23521444/e50b611a-ff33-11e6-8f40-0aa86728bdc7.png)

**Comment by SimonKrughoff** [comment]
I couldn't exactly match the scale and I'm in a different coordinate system, but here is the diffim:
![image](https://cloud.githubusercontent.com/assets/945715/23534353/40170a7c-ff6c-11e6-97c5-696248f57449.png)

Blue is detected.  The galaxy isn't actually detected as having different parity, that would be a different color.

As you can see, the whole area is a single footprint meaning that all the pixels are considered one object.  I think think the object position is just the maximum likelihood position.

Here are the same pixels smoothed with a 3 pixel Gaussian.  As you can see all the pixels in the smoothed image are quite high signal to noise in the footprint, so it passes the smell test for me.  This indicates that we need deblending to have any hope.  I've marked the diaSource from this diff (I only found one. I don't know why Bryce found more).  I reversed the color scale because it looked better.  I find the images are actually positive in the image I'm looking at.

![image](https://cloud.githubusercontent.com/assets/945715/23534547/86b96b4a-ff6d-11e6-8f5e-e14b528fae13.png)

**Comment by wmwv** [comment]
This is a fantastic discussion!

This is exactly the sort of issue that the Twinkles effort was made to bring up and explore!  Congratulations to all for getting to this point.

**Comment by jbkalmbach** [comment]
Oh, I showed the DIA sources from all the visits that's why I showed multiple. I actually only found one in this image as you did.

**Comment by drphilmarshall** [comment]
OK, this is all looking much better understood - thanks Bryce, thanks
Simon! (and Scott for the coordinate assistance :-).

It does seem like deblending would help here! Whats the outlook for this,
Simon?

One thing I wonder is:  are we *ensuring* that the DIASources come out
blended,  by choosing our DIASource association radius to be too large, or
something? We'd really like the DIASource characterizer to see 4 features
instead of 1, so if there are knobs we can turn to improve this, now's the
time to try them. Ideally I think we would err on the side of being
aggressive about *not* associating DIASources with existing DIAObjects
unless they were very very close (eg sep < 0.5 PSFwidth), but we'd need to
check that such aggression didn't lead to lots of spurious DIAObjects.

Thanks for pulling out the Objects and Sources as well, Bryce: if we can't
cope with blended DIASources then the shape of the DIAObjects over time
compared with the shape of the Object could become a key feature for lens
detection. Time delay measurement from the LSST products would be basically
impossible without returning to the pixels, though. Now that'd be an
interesting Monitor upgrade :-)

**Comment by SimonKrughoff** [comment]
Regarding blending in diffims, we have never even tried this.  I think it would be easy to turn on, but may not do what we want at all (i.e. what happens with negative peaks?).

Whether something is blended or not is a function only of the detection threshold.  The only way of making those peaks show up as different sources is by hand tuning the threshold to pick up the peaks, but not the saddle between them.  I believe that is the wrong thing to do.  I don't know when I'd be able to get to looking at deblending in diffims.  If someone wants to look at how this is done in `calibrate.py`, for example, it would be useful.

**Comment by drphilmarshall** [comment]
I agree about the need for caution when tuning detection thresholds. It's
odd that the DIASource characterization is so conservative though: in
Bryce's diffim there are clearly 4 well-separated point sources. Is the
DIASource detection threshold set conservatively so as to reduce the number
of bogus detections? If it has already been tuned to guard against this
then I don't think we can argue with that. However, if we are using some
default setting that has not been investigated before, there could be room
for experimentation. Who set the detection threshold? Could be worth
talking with them about this.

Re: deblending in diffims: is this in the DM plan already, or would it be
an addition to the Level 1 pipeline? I can imagine it being easy to turn on
but hard to tune...

**Comment by aprajita** [comment]
Hi @jbkalmbach, @SimonKrughoff & all,  I am one of the Strong Lensing Science Collaboration co-chairs. I realise this thread started quite some time ago, but I was wondering if I could use one or two these figs in a presentation for the PCW Blending workshop next week? I did mention some of this thread to Mario Juric and Robert Lupton at a workshop in May 2017 but I suspect things have moved on since then both with DM and your checks with DC2.  My understanding was (although I may not recall perfectly) that the object, and quasars, will be detected likely within the same footprint, and Robert thought the qsos should be in the catalogues. Mario also thought that level 1 deblending could be a possibility but am not sure if it is currently planned. Many thanks.
